lnwallet/btcwallet/btcwallet.go

```diff
@@ -1380,11 +1380,14 @@ func unminedTransactionsToDetail(
 // This will return transactions from the tip of the chain until the start
 // height (inclusive) and unconfirmed transactions. The account parameter serves
 // as a filter to retrieve the transactions relevant to a specific account. When
-// empty, transactions of all wallet accounts are returned.
+// empty, transactions of all wallet accounts are returned. A filter function
+// can also be specified to fetch transactions.
 //
 // This is a part of the WalletController interface.
 func (b *BtcWallet) ListTransactionDetails(startHeight, endHeight int32,
-	accountFilter string) ([]*lnwallet.TransactionDetail, error) {
+	accountFilter string,
+	filterFn func(tx *lnwallet.TransactionDetail) bool,
+) ([]*lnwallet.TransactionDetail, error) {
```

Comment:
Should we create a new function type alias in `lnwallet/interface.go` for this called `lnwallet.TxFilter` to shorten those definitions everywhere?

Reply:
I will see what I can do.